framework/audio/engine/iaudioengine.h

```diff
@@ -0,0 +1,63 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IAUDIOENGINE_H
+#define MU_AUDIO_IAUDIOENGINE_H
+
+#include <string>
+#include <functional>
+
+#include "modularity/imoduleexport.h"
+
+#include "iaudiosource.h"
+
+namespace mu {
+namespace audio {
+namespace engine {
+class IAudioEngine : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(IAudioEngine)
+
+public:
+    virtual ~IAudioEngine() = default;
+
+    using handle = unsigned int;
+    using time = float;
+
+    virtual bool init() = 0;
```

Comment:
It seems that these three methods are needed only in AudioEngineModule. Can we exclude them from that interface?

Reply:
There is a difficulty here that does not matter yet, but may arise in the future if we port the editor to the Web. At the moment, I can remove it from the interface.

Reply:
removed from interface